backend/app/controllers/spree/admin/quick_switch_controller.rb

```diff
@@ -0,0 +1,94 @@
+module Spree
+  module Admin
+    class QuickSwitchController < Spree::Admin::BaseController
+      layout false
+
+      def find_object
+        quick_switch_item = Spree::Backend::Config.quick_switch_items.detect do |item|
+          item.search_terms.include? searched_key.to_sym
+        end
+
+        if quick_switch_item
+          method(quick_switch_item.method).call
+        else
+          message = Spree.t("quick_switch.invalid_query")
+          respond_to do |format|
+            format.html { redirect_back(fallback_location: spree.admin_path, flash: { error: message }) }
```

Comment:
I did this so people could still hit this without JS (also what the controller's spec is doing). I can remove it and instead switch to a feature spec only.